include/fluent-bit/flb_lua.h

```diff
@@ -75,7 +75,7 @@ static inline int flb_lua_metadata_init(struct flb_lua_metadata *meta)
 /* convert from negative index to positive index */
 static inline int flb_lua_absindex(lua_State *l , int index)
 {
-#if defined(LUA_VERSION_NUM) && LUA_VERSION_NUM < 520
+#if defined(LUA_VERSION_NUM) && LUA_VERSION_NUM < 502
```

Comment:
I suggest to just remove this test and assume the true condition.

Fluent-bit uses a vendored version of LuaJit (that has LUA_VERSION_NUM defined as 501) and AFAICT doesn't allow the user to build against a custom version, so it is not useful having this kind of test since we already know its definitions.

Also, the test doesn´t appear to be correct, since if LUA_VERSION_NUM is not defined (which is not in older lua versions) it will fallback to lua_absindex path, which only exists in lua 5.2+